feat: viewable navigator in settings to testers and developers to view viewable feature flags

asalsys opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**
This PR is to show users the fetched feature flags that they pulled from the launch darkly api.
<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and accurately)
-->

CHANGELOG entry: added viewable navigator in settings to testers and developers to view release flag 

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-mobile/issues/21768

## **Manual testing steps**

```gherkin
Feature: Viewable Feature flags

  Scenario: user views feature flag dashboard
    Given basic functionality is toggled on

    When user navigates to settings Feature Flag Override
    Then they will see a dashboard to view the feature flags they have activated
```

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**
no feature flag override page as it did not exist
<!-- [screenshots/recordings] -->

### **After**

https://github.com/user-attachments/assets/08449226-15da-4e2d-97b5-e6cbcbe157f3



<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [X] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.


<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Adds a non‑production Feature Flag Override screen with override management, wired into Settings/navigation, backed by a new context, hook, and utilities, with tests and i18n updates.
> 
> - **UI**:
>   - Add `Views/FeatureFlagOverride` screen to list/search/filter flags and edit/reset overrides; displays stats via `useFeatureFlagStats`.
> - **Navigation/Settings**:
>   - Register `Routes.FEATURE_FLAG_OVERRIDE` and conditionally add `FeatureFlagOverride` screen and Settings drawer item (non‑production only).
> - **State/Context**:
>   - Introduce `FeatureFlagOverrideProvider` context for managing overrides (set/remove/clear, apply, count) and expose processed `FeatureFlagInfo` list.
>   - Wrap app root with `FeatureFlagOverrideProvider`.
> - **Hooks/Utils**:
>   - Add `useFeatureFlagStats` hook for flag type counts.
>   - Add feature‑flag utilities: `getFeatureFlagType`, `getFeatureFlagDescription`, `isMinimumRequiredVersionSupported`.
> - **Localization**:
>   - Add en strings for Settings “Feature Flag Override”.
> - **Tests**:
>   - Comprehensive tests for the new screen, context, hook, utilities, and updated snapshots.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit b80c37a21d207bc67193ea168dfacaf706d847cf. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

asalsys committed:
feat: create ui for local featureflag viewing and override

asalsys committed:
make flags values viewable

asalsys committed:
update the languages

asalsys committed:
update snapshots

asalsys committed:
add test coverage

asalsys committed:
fix test tsc

asalsys committed:
fix test tsc

asalsys committed:
fix boolean nested

asalsys committed:
add minimum version support indicator

asalsys committed:
fix local value edge case

asalsys committed:
revert: remove changes from all language files except en.json

tommasini commented on app/components/Nav/Main/MainNavigator.js:
```diff
@@ -1132,6 +1133,13 @@ const MainNavigator = () => {
           ...GeneralSettings.navigationOptions,
         }}
       />
+      <Stack.Screen
```
Should we only initialize this stack if not production?

tommasini commented on app/contexts/README.md:
```diff
@@ -0,0 +1,133 @@
+# Feature Flag Override Context
```
I dont think we need this documentation for the feature, its self explanatory, wdyt?

If you think otherwise, we should change the name of the file to reflect what it is

sallem-consensys commented on app/hooks/useFeatureFlagStats.test.ts:
```diff
@@ -0,0 +1,456 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
enabled rule again

asalsys committed:
address comments to adhere to ts rules

asalsys committed:
fix typing

asalsys committed:
fix tsc

asalsys committed:
flag dashboard only on non production builds

asalsys committed:
hide feature in production

asalsys committed:
fix lint and format

asalsys committed:
update snapshots

asalsys committed:
fix tests

asalsys committed:
fix test

sallem-consensys commented on locales/languages/de.json:
```diff
@@ -1865,6 +1865,7 @@
     "token_is_needed_to_continue": "{{tokenSymbol}} ist erforderlich, um fortzufahren",
     "fund_your_wallet": "Fund your wallet",
     "get_ready_for_web3": "Get your wallet ready to use web3",
+    "fund_your_wallet_to_get_started": "Zahlen Sie Gelder auf Ihre Wallet ein, um bei web3 loszulegen.",
```
yes im not sure how it got pulled in from updating the merge conflict.

asalsys committed:
revert: remove locale changes except en.json

- Reverted all language files except en.json to main branch state
- Kept only the feature_flag_override additions in en.json

sallem-consensys commented on app/util/feature-flags/index.ts:
```diff
@@ -0,0 +1,86 @@
+  }
+  return 'string';
+};
+
```
yes I will add documentation as part of another ticket for feature flag refinement

sallem-consensys commented on app/components/Views/FeatureFlagOverride/FeatureFlagOverride.tsx:
```diff
@@ -0,0 +1,455 @@
+            thumbColor={theme.brandColors.white}
+            ios_backgroundColor={theme.colors.border.muted}
+          />
+        );
```
this PR is for read only currently
